pennylane/plugins/tests/__init__.py

```diff
@@ -47,4 +32,205 @@
 is set to a high enough value to account for stochastic fluctuations. Flaky is used to automatically
 repeat failed tests.
+
+There are several methods for running the tests against a particular device (i.e., for
+``'default.qubit'``), detailed below.
+
+Using pytest
+------------
+
+.. code-block:: console
+
+    pytest path_to_pennylane_src/plugins/tests --device=default.qubit --shots=10000 --analytic=False
+
+The location of your PennyLane installation may differ depending on installation method and
+operating system. To find the location, you can use the :func:`~.get_device_tests` function:
+
+>>> from pennylane.plugins.tests import get_device_tests
+>>> get_device_tests()
+
+The pl-device-test CLI
+----------------------
+
+Alternatively, PennyLane provides a command line interface for invoking the device tests.
+
+.. code-block:: console
+
+    pl-device-test --device default.qubit --shots 10000 --analytic False
+
+Within Python
+-------------
+
+Finally, the tests can be invoked within a Python session via the :func:`~.test_device`
+function:
+
+>>> from pennylane.plugins.tests import test_device
+>>> test_device("default.qubit")
+
+For more details on the available arguments, see the :func:`~.test_device` documentation.
+
+Functions
+---------
 """
+# pylint: disable=import-outside-toplevel,too-many-arguments
+import argparse
+import pathlib
+import subprocess
+import sys
+
+
+# determine if running in an interactive environment
+import __main__
+
+interactive = False
+
+try:
+    __main__.__file__
+except AttributeError:
+    interactive = True
+
+
+def get_device_tests():
+    """Returns the location of the device integration tests."""
+    return str(pathlib.Path(__file__).parent.absolute())
+
+
+def test_device(
+    device, analytic=None, shots=None, skip_ops=True, flaky_report=False, pytest_args=None, **kwargs
+):
+    """Run the device integration tests using an installed PennyLane device.
+
+    Args:
+        device (str): the name of the device to test
+        analytic (bool): Whether to run the device in analytic mode (where
+            expectation values and probabilities are computed exactly from the quantum state)
+            or non-analytic/"stochastic" mode (where probabilities and expectation
+            values are *estimated* using a finite number of shots.)
+            If not provided, the device default is used.
+        shots (int): The number of shots/samples used to estimate expectation
+            values and probability. Only takes affect if ``analytic=False``. If not
+            provided, the device default is used.
+        skip_ops (bool): whether to skip tests that use operations not supported
+            by the device
+        pytest_args (list[str]): additional PyTest arguments and flags
+        **kwargs: Additional device keyword args
+
+    **Example**
+
+    >>> from pennylane.plugins.tests import test_device
+    >>> test_device("default.qubit")
```

Comment:
Worked great when ran from an `ipython` session or as part of a Python script! :slightly_smiling_face: :100: 

In a Jupyter notebook the output was directed into the terminal where `jupyter notebook` was invoked (and there was no output in the notebook itself).

Reply:
Good catch about Jupyter notebook! Do you think this is a common use-case we should support? I think we should not support running tests from within a Jupyter notebook

Reply:
Kind of on the fence here :thinking: I think it's not too crucial, though if interactive sessions are supported then a Jupyter notebook would also be an important case. Having said that I think it's not a major point